stan/math/prim/fun/logit.hpp

```diff
@@ -80,11 +80,27 @@ struct logit_fun {
  * @param x container
  * @return elementwise logit of container elements
  */
-template <typename T>
+template <typename T,
+          require_not_container_st<is_container, std::is_arithmetic, T>...>
 inline auto logit(const T& x) {
   return apply_scalar_unary<logit_fun, T>::apply(x);
 }
 
+/**
+ * Version of logit() that accepts Eigen Matrix/Array objects or expressions.
+ *
+ * @tparam T Type of x
+ * @param x Eigen Matrix/Array or expression
+ * @return the logit of each variable in the container.
+ */
+template <typename T,
+          require_container_st<is_container, std::is_arithmetic, T>...>
+inline auto logit(const T& x) {
+  return apply_vector_unary<T>::apply(x, [&](const auto& v) {
+    return (v.derived().array() / (1 - v.derived().array())).log();
```

Comment:
This one should use `Eigen::Ref` so as not to evaluate ˙`v` twice.

Reply:
Does this need & in the lambda? And why `const auto&` We don't reuse v in the lambda so I think we could just  `auto&&`

Reply:
@andrjohns Sorry, I overlooked this reply. I disagree with @SteveBronder here. In my opinion we should always pass arguments as `const&` if possible, to make it obvious the function does not change them. `auto&&` or any other forwarding reference should only be used when we actually need perfect forwarding.